daffodil-schematron/src/main/scala/org/apache/daffodil/validation/schematron/SchematronResult.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.validation.schematron
+
+import org.apache.daffodil.api.RawValidationResult
+import org.apache.daffodil.api.ValidationFailure
+import org.apache.daffodil.api.ValidationResult
+import org.apache.daffodil.api.ValidationWarning
+
+import java.util
+
+/**
+ * Captures the output of Schematron validation as a Daffodil ValidationResult
+ * @param warnings Schematron warnings parsed into ValidationWarning objects
+ * @param errors Schematron errors parsed into ValidationFailure objects
+ * @param svrl Full SVRL output
+ */
+case class SchematronResult(warnings: util.Collection[ValidationWarning],
+                            errors: util.Collection[ValidationFailure],
+                            svrl: String) extends ValidationResult with RawValidationResult {
+
+  def rawValidationData(): Array[Byte] = svrl.getBytes
+}
```

Comment:
The validators are pluggable and assuming other implementations could have raw data that would be useful to write then you wouldnt want to have to wire those other projects into the CLI explicitly to allow that.

> I guess my concern is that that a user of this schemtron validator might want access to the Svrl String, or maybe even the svrl XML Document, but instead we're converting it to bytes just to be generic and forcing them to do something with it. 

With this PR you can do either.

The raw trait uses bytes to be generic in handling _raw_ output across all possible validator implementations, since we dont know what an implementation in the wild might spit out.  The only raw consumer right now is the CLI where it is just dumping bytes to a file, and we use the raw trait to avoid depending on the schematron project explicitly.

You can still extract SVRL without using the raw trait, but to do so you have to explicitly depend on the schematron implementation.  You could follow this pattern with any validator and consumer.

You can see example implementations of this in the new CLI tests in this PR.

Another approach to avoid the second trait would be to roll it into the ValidationResult trait providing a default empty raw property.  I thought about that when implementing this, but an empty raw property in the main trait seemed less appealing than another trait.

Reply:
> You can see example implementations of this in the new CLI tests in this PR.

Uhm, well maybe not, but I can add some.  I must of been thinking of the SAPI test that does this.  But doesnt illustrate the point as well as a schematron test in the CLI would.

Reply:
I see what you mean - lines 906-917 in Main.scala performs a match on ValidationResult to get a case Some(vr: RawValidationResult) and writes out vr.rawValidationData() accordingly.  I also think there would be nothing to stop someone from doing a match on ValidationResult to get a case Some(vr: MyCustomValidationResult) if they want to use some custom data without conversion to/from raw bytes, right?  So people are free to use both ways in their own custom validator implementation - extend RawValidatorResult to let the CLI write the file and also provide custom methods in MyCustomValidationResult, which seems fair to me.

Reply:
@tuxji, right on :+1: 

> there would be nothing to stop someone from doing a match on ValidationResult to get a case Some(vr: MyCustomValidationResult 

The only thing stopping you would be having a dependency to the artifact that contains MyCustomValidationResult.  Which in the case of the SchematronResult and the CLI there is no dependency there, but the CLI only needs the API level raw type.